### ISSUE TITLE
Multiple connections to the same host fail with "Cannot recconect. Please reload the window."

### ISSUE THREAD

**Issue by jungaretti** [microsoft/vscode-remote-release#2621]
- VSCode Version: 1.43.2
- Local OS Version: macOS 10.15.4
- Remote OS Version: Ubuntu 18.04.4 LTS
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Remove `.vscode-server` on host
2. Remote-SSH: Connect to Host...
3. username@lab01.university.edu
5. Close Remote Connection
6. Remote-SSH: Connect to Host...
7. username@lab01.university.edu

VSCode fails to connect with the error "Cannot reconnect. Please reload this window." Interestingly, if I replace step 7 with username@**lab02**.university.edu, everything works as expected!

I also notice that output always contains `Found running server...` when the connection fails. It seems like the error has something to do with a server running on a specific machine.

I'm using the default settings for Remote - SSH. I've tried individually toggling Enable Agent Forwarding, Lockfiles in Tmp, and Use Local Server. None of these settings seem to fix the error. The only way to resolve the error is to remove `.vscode-server`.

Does this issue occur when you try this locally?: Issue involves connecting to remote
Does this issue occur when you try this locally and all extensions are disabled?: Issue occurs when all other extensions are disabled

```
[23:49:51.209] Log Level: 2
[23:49:51.211] remote-ssh@0.50.1
[23:49:51.211] darwin x64
[23:49:51.213] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a22726f636b686f707065722d30322e63732e776973632e656475222c2275736572223a22756e67617265747469227d", attempt 1
[23:49:51.213] SSH Resolver called for host: ungaretti@rockhopper-02.cs.wisc.edu
[23:49:51.213] Setting up SSH remote "rockhopper-02.cs.wisc.edu"
[23:49:51.217] Acquiring local install lock: /var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/vscode-remote-ssh-ungaretti@rockhopper-02.cs.wisc.edu-install.lock
[23:49:51.231] Looking for existing server data file at /Users/jungaretti/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-ungaretti@rockhopper-02.cs.wisc.edu-0ba0ca52957102ca3527cf479571617f0de6ed50-0.50.1/data.json
[23:49:51.233] Using commit id "0ba0ca52957102ca3527cf479571617f0de6ed50" and quality "stable" for server
[23:49:51.238] Install and start server if needed
[23:49:51.246] Checking ssh with "ssh -V"
[23:49:51.273] > OpenSSH_8.1p1, LibreSSL 2.7.3
[23:49:51.276] askpass server listening on /var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/vscode-ssh-askpass-490a9ff2f0f0c98e584f9c4549be00cc4aa00db8.sock
[23:49:51.276] Spawning local server with {"ipcHandlePath":"/var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/vscode-ssh-askpass-e56623203d551a51d9eeea0942f68a8947381e01.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","54326","-o","ConnectTimeout=15","ungaretti@rockhopper-02.cs.wisc.edu"],"dataFilePath":"/Users/jungaretti/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-ungaretti@rockhopper-02.cs.wisc.edu-0ba0ca52957102ca3527cf479571617f0de6ed50-0.50.1/data.json"}
[23:49:51.276] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/jungaretti/.vscode/extensions/ms-vscode-remote.remote-ssh-0.50.1/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/private/var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/AppTranslocation/156629D0-73E5-4FAB-A868-188589F310E9/d/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)","VSCODE_SSH_ASKPASS_MAIN":"/Users/jungaretti/.vscode/extensions/ms-vscode-remote.remote-ssh-0.50.1/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/vscode-ssh-askpass-490a9ff2f0f0c98e584f9c4549be00cc4aa00db8.sock"}
[23:49:51.280] Spawned 13701
[23:49:51.383] > local-server> Spawned ssh: 13702
[23:49:51.386] stderr> OpenSSH_8.1p1, LibreSSL 2.7.3
[23:49:51.635] stderr> debug1: Server host key: ssh-rsa SHA256:6eEAzNpxMssGmSZLGDBylby4V0bx3RDgfBd9uVl3cF8
[23:49:51.951] Got askpass request: {"request":"Password:"}
[23:49:51.952] Showing password prompt
[23:49:51.954] Listening for interwindow password on /var/folders/qd/hyq_t8vn6n59685rsvb_0lz40000gn/T/vscode-ssh-askpass-3e02261cebf0a01cda3284206db3c57a01900544.sock
[23:49:51.954] Writing password prompt to globalState
[23:49:56.972] Got password response
[23:49:56.972] Interactor gave response: ******************
[23:49:56.973] Cleaning up other-window auth server
[23:49:57.828] stderr> Authenticated to rockhopper-02.cs.wisc.edu ([128.105.37.192]:22).
[23:49:58.054] > Welcome to Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-91-generic x86_64)
> 
> 2 packages can be updated.
> 0 updates are security updates.
> 
> 
> ===============================================================================
>       This instructional workstation is installed with Ubuntu 18.04.4 LTS
> 
>  If you encounter any issues, please send a problem report to lab@cs.wisc.edu
> ===============================================================================
>       REMINDER: NO FOOD or DRINK IN THE CS INSTRUCTIONAL COMPUTER LABS
>             NEVER POWER DOWN WORKSTATIONS IN THE COMPUTER LABS
> ===============================================================================
[23:49:58.090] > ready: 2f3be5e246c5
[23:49:58.127] > Linux 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020
[23:49:58.127] Platform: linux
[23:49:58.219] > 2f3be5e246c5: running
[23:49:58.235] > Acquiring lock on /u/u/n/ungaretti/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.ungaretti.0ba0ca52957102ca3527cf479571617f0de6ed50
[23:49:58.237] > \ln /u/u/n/ungaretti/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.ungaretti.0ba0ca52957102ca3527cf479571617f0de6ed50.target /u/u/n/ungaretti/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.ungaretti.0ba0ca52957102ca3527cf479571617f0de6ed50
[23:49:58.241] > Found existing installation at /u/u/n/ungaretti/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50...
[23:49:58.250] > Found running server...
>  
> *
> * Reminder: You may only use this software with Visual Studio family products,
[23:49:58.251] > * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[23:49:58.258] > Checking server status on port 34615 with wget
[23:49:58.260] > 2f3be5e246c5: start
> sshAuthSock====
> agentPort==34615==
> osReleaseId==ubuntu==
[23:49:58.261] > arch==x86_64==
> webUiAccessToken====
> tmpDir==/run/user/3155==
> platform==linux==
> 2f3be5e246c5: end
[23:49:58.262] Received install output: 
sshAuthSock====
agentPort==34615==
osReleaseId==ubuntu==arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/3155==
platform==linux==

[23:49:58.262] Remote server is listening on port 34615
[23:49:58.262] Parsed server configuration: {"agentPort":34615,"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/3155","platform":"linux"}
[23:49:58.263] Persisting server connection details to /Users/jungaretti/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-ungaretti@rockhopper-02.cs.wisc.edu-0ba0ca52957102ca3527cf479571617f0de6ed50-0.50.1/data.json
[23:49:58.265] Starting forwarding server. localPort 54334 -> socksPort 54326 -> remotePort 34615
[23:49:58.265] Forwarding server listening on 54334
[23:49:58.265] Waiting for ssh tunnel to be ready
[23:49:58.266] [Forwarding server 54334] Got connection 0
[23:49:58.267] Tunneled remote port 34615 to local port 54334
[23:49:58.267] Resolved "ssh-remote+7b22686f73744e616d65223a22726f636b686f707065722d30322e63732e776973632e656475222c2275736572223a22756e67617265747469227d" to "127.0.0.1:54334"
[23:49:58.277] ------




[23:49:58.319] [Forwarding server 54334] Got connection 1
[23:49:58.319] [Forwarding server 54334] Got connection 2
```

**Comment by roblourens** [comment]
Is this log from an instance when it failed? So does it initially connect, then get disconnected, and fail to reconnect? That is basically the scenario when you would see that "Cannot reconnect" message but the reconnect attempt doesn't show up in the log.

**Comment by jungaretti** [comment]
Yes, this is the output of an instance when it failed to connect. It’s possible that it gets disconnected, but I’m not sure why that would happen. It consistently works on the first ever connection (without a .vscode-server installation), and consistently fails on subsequent connections.

Our university has many machines that we can remotely access. I’m not sure what the technical term for this is, but I can access my home directory form any of these machines. If I replace step 7 from my original post with a connection to a different machine (but accessing the same .vscode-server installation), everything works as expected.

**Comment by roblourens** [comment]
I'm not sure why it's failing to connect to the remote server that it started previously. I would imagine that running "Kill VS Code Server" after step 4 then connecting to the same server again would work. Maybe some administrative setup on the remote is interfering with the port that the remote server exposes.

**Comment by jungaretti** [comment]
Yup, your imagination is correct! Using the "Kill VS Code Server" command after step 4 fixed the problem. Thanks, I appreciate your followup.

**Comment by roblourens** [comment]
Sorry I don't have a better solution. You might ask a system administrator whether they have any ideas. The server on the remote listens on some random port, then we use dynamic port forwarding over ssh to forward that port.

**Comment by Jinyong-Huang** [comment]
I have same question.


[09:17:19.680] > 3810a164b616: start
> sshAuthSock====
> agentPort==45409==
> osReleaseId==centos==
> arch==x86_64==
> webUiAccessToken====
> tmpDir==/run/user/0==
> 3810a164b616: end
> 
[09:17:19.681] Received install output: 
sshAuthSock====
agentPort==45409==
osReleaseId==centos==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/0==

[09:17:19.681] Remote server is listening on port 45409
[09:17:19.681] Parsed server configuration: {"agentPort":45409,"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/0"}
[09:17:19.683] Starting forwarding server. localPort 54620 -> socksPort 54612 -> remotePort 45409
[09:17:19.684] Forwarding server listening on 54620
[09:17:19.684] Waiting for ssh tunnel to be ready
[09:17:19.687] [Forwarding server 54620] Got connection 0
[09:17:19.691] Tunneled remote port 45409 to local port 54620
[09:17:19.691] Resolved "ssh-remote+10.0.14.49" to "127.0.0.1:54620"
[09:17:19.699] ------




[09:17:19.729] [Forwarding server 54620] Got connection 1
[09:17:19.733] [Forwarding server 54620] Got connection 2
[09:17:45.960] ------
